### PR TITLE
chore: split JUnit4/Jupiter testkit classes

### PR DESCRIPTION
**Pull request by octonato**
As discussed here:
https://github.com/lightbend/kalix-jvm-sdk/pull/1876#discussion_r1402146911

Not marking `KalixTestKitResource` as deprecated for now. To move in that direction, we should foresee a migration page.

### CODE REVIEW

File: sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/KalixTestKitResource.java [hunk 1]
```diff
@@ -41,14 +41,14 @@
...
- * import kalix.javasdk.testkit.junit.KalixTestKitResource;
+ * import kalix.javasdk.testkit.junit.jupiter.KalixTestKitExtension;
  *
  * public class MyKalixIntegrationTest {
  *
  *   private static final Kalix MY_KALIX = new Kalix(); // with registered services
  *
  *   &#64;ClassRule
- *   public static final KalixTestKitResource testKit = new KalixTestKitResource(MY_KALIX);
+ *   public static final KalixTestKitExtension testKit = new KalixTestKitExtension(MY_KALIX);
```
Comment (ennru) [inline sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/KalixTestKitResource.java:51]: This looks wrong, this is the `KalixTestKitResource` source file.

File: sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/jupiter/KalixTestKitExtension.java [hunk 2]
```diff
@@ -16,99 +16,190 @@
...
-import org.junit.platform.commons.util.ReflectionUtils;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.rules.ExternalResource;
 
-import java.lang.reflect.Field;
+/**
+ * A JUnit external resource for {@link KalixTestKit}, which automatically manages the lifecycle of
```
Comment (ennru) [inline sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/jupiter/KalixTestKitExtension.java:36]:
```suggestion
 * A JUnit Extension for {@link KalixTestKit}, which automatically manages the lifecycle of
```

File: sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/KalixTestKitResource.java (hunk 1 above)
Comment (octonato) [reply sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/KalixTestKitResource.java:51]: yep, too many find-and-replace

File: sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/jupiter/KalixTestKitExtension.java [hunk 3]
```diff
@@ -16,99 +16,190 @@
...
+ * the testkit. The testkit will be automatically stopped when the test completes or fails.
+ *
+ * <p><b>Note</b>: JUnit is not provided as a transitive dependency of the Java SDK testkit module
+ * but must be added explicitly to your project.
```
[the remark this answers is not in the document]
Comment (octonato) [reply sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/jupiter/KalixTestKitExtension.java:40]: We will have it in the KalixTestKitResource as well. And we should remove it. It's now being added as transitive dependency.

File: sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/KalixTestKitResource.java [hunk 4]
```diff
@@ -26,18 +26,12 @@
...
 import kalix.javasdk.testkit.EventingTestKit.Topic;
 import kalix.javasdk.testkit.KalixTestKit;
-import org.junit.jupiter.api.extension.AfterAllCallback;
-import org.junit.jupiter.api.extension.BeforeAllCallback;
-import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.rules.ExternalResource;
 
 /**
  * A JUnit external resource for {@link KalixTestKit}, which automatically manages the lifecycle of
  * the testkit. The testkit will be automatically stopped when the test completes or fails.
```
Comment (ennru) [inline sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/KalixTestKitResource.java:33]:
```suggestion
 * A JUnit 4 external resource for {@link KalixTestKit}, which automatically manages the lifecycle of
 * the testkit. The testkit will be automatically stopped when the test completes or fails.
 *
 * If you prefer JUnit 5 "Jupiter" use {@link KalixTestKitExtension}.
```

File: sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/jupiter/KalixTestKitExtension.java [hunk 5]
```diff
@@ -16,99 +16,187 @@
...
-import org.junit.platform.commons.util.ReflectionUtils;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.rules.ExternalResource;
 
-import java.lang.reflect.Field;
+/**
+ * A JUnit Extension for {@link KalixTestKit}, which automatically manages the lifecycle of
```
Comment (ennru) [inline sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/junit/jupiter/KalixTestKitExtension.java:36]:
```suggestion
 * A JUnit 5 "Jupiter" Extension for {@link KalixTestKit}, which automatically manages the lifecycle of
```

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
This fixes 
- #41

**Pull request merged**